### PR TITLE
fix: allow providing None for attachments to clear attachments when editing a message

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

Providing None when editing a message should now delete all attachments,
rather than requiring a empty list.
## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/interactions/base.py [hunk 1]
```diff
@@ -370,9 +370,13 @@ async def edit_original_message(
             A list of attachments to keep in the message. If ``[]`` is passed
             then all existing attachments are removed.
             Keeps existing attachments if not provided.
+            Can also pass ``None`` to clear all attachments.
```
Comment (shiftinv) [inline disnake/interactions/base.py:373]:
```suggestion
            A list of attachments to keep in the message.
            If ``[]`` or ``None`` is passed then all existing attachments are removed.
            Keeps existing attachments if not provided.
```

Comment (onerandomusername) [reply disnake/interactions/base.py]: https://github.com/DisnakeDev/disnake/pull/457/commits/c65c8b80854ead90ebee1ebc00cbc70aaf70be10

File: disnake/interactions/base.py [hunk 2]
```diff
@@ -397,12 +397,15 @@ async def edit_original_message(
             Files will be appended to the message, see the ``attachments`` parameter
             to remove/replace existing files.
         attachments: List[:class:`Attachment`]
```
Comment (shiftinv) [inline disnake/interactions/base.py:399]:
```suggestion
        attachments: Optional[List[:class:`Attachment`]]
```
same thing for other `attachments` parameters

File: disnake/webhook/async_.py [hunk 3]
```diff
@@ -1719,12 +1723,15 @@ async def edit_message(
             .. versionadded:: 2.0
 
         attachments: List[:class:`Attachment`]
-            A list of attachments to keep in the message. If ``[]`` is passed
-            then all existing attachments are removed.
+            A list of attachments to keep in the message.
+            If ``[]`` or ``None`` is passed then all existing attachments are removed.
```
Comment (shiftinv) [inline disnake/webhook/async_.py:1727]: parameter annotation not updated

File: disnake/webhook/async_.py [hunk 4]
```diff
@@ -784,10 +784,14 @@ async def edit(
...
```
Comment (shiftinv) [inline disnake/webhook/async_.py:788]: parameter annotation not updated

File: disnake/webhook/sync.py [hunk 5]
```diff
@@ -439,10 +439,14 @@ def edit(
             Files will be appended to the message, see the ``attachments`` parameter
             to remove/replace existing files.
         attachments: List[:class:`Attachment`]
-            A list of attachments to keep in the message. If ``[]`` is passed
-            then all existing attachments are removed.
+            A list of attachments to keep in the message.
+            If ``[]`` or ``None`` is passed then all existing attachments are removed.
```
Comment (shiftinv) [inline disnake/webhook/sync.py:443]: parameter annotation not updated

File: disnake/interactions/base.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/interactions/base.py]: https://github.com/DisnakeDev/disnake/pull/457/commits/3dc30b1af1fdbfd9fddbd3ab07370b36cae42029

File: disnake/webhook/async_.py (hunk 4 above)
Comment (onerandomusername) [reply disnake/webhook/async_.py:788]: https://github.com/DisnakeDev/disnake/pull/457/commits/3dc30b1af1fdbfd9fddbd3ab07370b36cae42029

File: disnake/webhook/sync.py (hunk 5 above)
Comment (onerandomusername) [reply disnake/webhook/sync.py:443]: https://github.com/DisnakeDev/disnake/pull/457/commits/3dc30b1af1fdbfd9fddbd3ab07370b36cae42029

File: disnake/webhook/async_.py (hunk 3 above)
Comment (onerandomusername) [reply disnake/webhook/async_.py:1727]: https://github.com/DisnakeDev/disnake/pull/457/commits/3dc30b1af1fdbfd9fddbd3ab07370b36cae42029

File: disnake/webhook/async_.py [hunk 6]
```diff
@@ -783,11 +783,15 @@ async def edit(
...
+        attachments: Optional[List[:class:`Attachment`]]
+            A list of attachments to keep in the message.
+            If ``[]`` or ``None`` is passed then all existing attachments are removed.
             Keeps existing attachments if not provided.
 
+            .. versionchanged:: 2.5
+                Supports passing ``None`` to clear attachments.
+
+
             .. versionadded:: 2.2
```
Comment (shiftinv) [inline disnake/webhook/async_.py:795]:
```suggestion
            .. versionadded:: 2.2

            .. versionchanged:: 2.5
                Supports passing ``None`` to clear attachments.
```

File: disnake/webhook/sync.py [hunk 7]
```diff
@@ -438,11 +438,15 @@ def edit(
...
+        attachments: Optional[List[:class:`Attachment`]]
+            A list of attachments to keep in the message.
+            If ``[]`` or ``None`` is passed then all existing attachments are removed.
             Keeps existing attachments if not provided.
 
+            .. versionchanged:: 2.5
+                Supports passing ``None`` to clear attachments.
+
+
             .. versionadded:: 2.2
```
Comment (shiftinv) [inline disnake/webhook/sync.py:450]:
```suggestion
            .. versionadded:: 2.2

            .. versionchanged:: 2.5
                Supports passing ``None`` to clear attachments.
```

**Review by shiftinv — APPROVED** [review]
lgtm

**Pull request merged**